src/main/scala/akka/persistence/cassandra/query/journal/scaladsl/EventsByTagPublisher.scala

```diff
@@ -0,0 +1,94 @@
+/*
+package akka.persistence.cassandra.query.journal
+
+import akka.persistence.cassandra.query.EventsByTagFetcher
+import akka.persistence.cassandra.query.EventsByTagPublisher.TaggedEventEnvelope
+import akka.persistence.cassandra.query.EventsByTagPublisher.TaggedEventEnvelope
+import akka.persistence.query.EventEnvelope
+import akka.pattern.ask
+import java.time.LocalDate
+import java.time.ZoneOffset
+import java.util.UUID
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+import akka.actor.ActorLogging
+import akka.actor.DeadLetterSuppression
+import akka.actor.NoSerializationVerificationNeeded
+import akka.actor.OneForOneStrategy
+import akka.actor.Props
+import akka.actor.SupervisorStrategy
+import akka.persistence.cassandra.journal.CassandraJournal
+import akka.persistence.query.EventEnvelope
+import akka.stream.actor.ActorPublisher
+import akka.stream.actor.ActorPublisherMessage.Cancel
+import akka.stream.actor.ActorPublisherMessage.Request
+import com.datastax.driver.core.PreparedStatement
+import com.datastax.driver.core.Session
+import com.datastax.driver.core.utils.UUIDs
+
+class EventsByTagPublisher(
```

Comment:
Just an attempt at EventsByTagPublisher using @patriknw's branch. It should be possible to reuse QueryActorPublisher to avoid duplication.

This use case is much simpler than LevelDb, because we don't need to worry about push updates (all is pull from database). So the use case fits the abstraction very well. This attempt is not finished, becasue the EventsByTagFetcher is quite tightly (and in my opinion not transparently and type safely) coupled with the publisher and manages the queries. We refactor that to fit in the (State, Max) => Future[Vector[T]] (or even (State, Max) => Future[T]) easily though.